### ISSUE TITLE
[HTML5] Objects changing depth every frame causes gradual frame drop

### ISSUE THREAD

**Issue by tabularelf** [YoYoGames/GameMaker-Bugs#7629]
### Description
I had noticed this many months ago when I was helping someone with their game. For some unknown reason, their game was gradually dropping in frames. The only eventual cause I found was by commenting out every single depth change, that the game stopped dropping in frames all together.

At the time, I wasn't able to replicate this at all on other platforms, not even in a separate test project. As this is not my own project, I have gotten explicit permission to submit this at the current state it was in.

Note: This does happen in every level I've witnessed, but only some levels demonstrate it faster.

### Steps To Reproduce
1. Run game on HTML5
2. Press M twice to get to the second level. (In the project it is labelled as `r04`)
3. Sit & wait, watching the FPS and instance count from top left.
4. Notice frame drop, despite sitting still and doing nothing.
5. Comment out every instance of `depth =` change
6. Repeat 1-3. No frame drop

### Which version of GameMaker are you reporting this issue for?
IDE v2024.8.0.169 Runtime v2024.8.0.216

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
HTML5

[c7f4d6cd-b649-4615-8873-47b2d27d14c7](https://api.gamemaker.io/api/github/downloads/c7f4d6cd-b649-4615-8873-47b2d27d14c7)

**Comment by tabularelf** [comment]
Attached below is a video that demonstrates the exact issue, you'll notice that the framerate will drop despite doing "nothing".
Note: I have thoroughly checked to make sure that no additional instances were being made.

https://github.com/user-attachments/assets/a3889853-e2dc-4fef-8531-d7ca4c2a1fd6

**Comment by tabularelf** [comment]
I have made attempts at replicating it, but the closest I've been able to depict is that changing event in TWO places (like this project is doing) seems to cause an immediate drop in frame rate with a separate project. But not a gradual frame drop, like this game project is showcasing. Something is amiss here, but this could give some further insight.

In the sample I'm uploading above, depth is change for all instances of `obj_test` to `0`, and then each instance changes its own depth to `2`, causes the frame drop to be more instant. Though I suspect that is its own separate issue going abou it this way. In either case, needs more investigating.
[html5_depth_bug.zip](https://github.com/user-attachments/files/16993312/html5_depth_bug.zip)
